agent/kvs_endpoint.go

```diff
@@ -72,7 +72,7 @@ func (s *HTTPHandlers) KVSGet(resp http.ResponseWriter, req *http.Request, args
 
 	// Make the RPC
 	var out structs.IndexedDirEntries
-	if err := s.agent.RPC(method, &args, &out); err != nil {
+	if err := s.agent.RPC(method, args, &out); err != nil {
```

Comment:
> 2\. After applying the rpcHoldTimeout to all client RPC calls, I noticed one type of blocking query (`KVS.Get` / `KVS.List`) wasn't having its timeout set properly based on the query parameters. Removing an ampersand from the `args` parameter fixed this issue ([92525c4](https://github.com/hashicorp/consul/commit/92525c40ee6feefcd0b299af16095c34a2971adb)). This probably also affects the existing RPC retry logic for this RPC method so could possibly be extracted as a separate fix PR.

This seems like a really weird class of bug as the behaviour shouldn't depend on `args` being mutated during RPC. @wjordan was this encountered during your own testing?

Reply:
Just as precaution I'd like to revert this one-liner and file another issue regarding KVS in case we ever need to debug this PR in the future.

Reply:
Yes, I discovered this issue in my own testing on a production cluster, without this change I was seeing timeouts on blocking queries to these methods. I can't say I quite understand why this change fixed the issue, only that it did seem to fix it for my particular setup.

Moving the fix to a separate PR sounds like a good precaution in any case.